packages/@aws-cdk/aws-ec2/README.md

```diff
@@ -173,29 +173,40 @@ Application subnets will route to the NAT Gateway.
 
 ### Allowing Connections
 
-In AWS, all connections to and from EC2 instances are governed by *Security
-Groups*. You can think of these as a firewall with rules. All Constructs that
-create instances on your behalf implicitly have such a security group.
-Unless otherwise indicated using properites, the security groups start out
-empty; that is, no connections are allowed by default.
-
-In general, whenever you link two Constructs together (such as the load balancer and the
-fleet in the previous example), the security groups will be automatically updated to allow
-network connections between the indicated instances. In other cases, you will need to
-configure these allows connections yourself, for example if the connections you want to
-allow do not originate from instances in a CDK construct, or if you want to allow
-connections among instances inside a single security group.
-
-All Constructs with security groups have a member called `connections`, which
-can be used to configure permissible connections. In the most general case, a
-call to allow connections needs both a connection peer and the type of
-connection to allow:
+In AWS, all network traffic in and out of **Elastic Network Interfaces** (ENIs)
+is controlled by **Security Groups**. You can think of Security Groups as a
+firewall with a set of rules. The Security Groups start out empty (that is,
+no network traffic is allowed by default) and rules need to be added to them
```

Comment:
Turns out there is a default rule that is added if a Security Group is not explicitly configured with any egress rules. The following is mentioned in the [CloudFormation documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group.html#w2ab1c21c10d476c17). As a suggested workaround, I've been able to achieve a locked down SecurityGroup by adding the below mentioned limit-egress-to-localhost rule where needed.
> **Remove Default Rule**
When you create a VPC security group, Amazon EC2 creates a default egress rule that allows egress traffic on all ports and IP protocols to any location. The default rule is removed only when you specify one or more egress rules. If you want to remove the default rule and limit egress traffic to just the localhost (127.0.0.1/32), use the following example.